tsdb/engine/tsm1/reader.go

```diff
@@ -202,27 +202,27 @@ func NewTSMReader(f *os.File) (*TSMReader, error) {
 }
 
 func (t *TSMReader) applyTombstones() error {
-	// Read any tombstone entries if the exist
-	tombstones, err := t.tombstoner.ReadAll()
-	if err != nil {
-		return fmt.Errorf("init: read tombstones: %v", err)
-	}
-
-	if len(tombstones) == 0 {
-		return nil
-	}
-
 	var cur, prev Tombstone
-	cur = tombstones[0]
-	batch := []string{cur.Key}
-	for i := 1; i < len(tombstones); i++ {
-		cur = tombstones[i]
-		prev = tombstones[i-1]
-		if prev.Min != cur.Min || prev.Max != cur.Max {
+	batch := make([]string, 0, 4096)
```

Comment:
@sebito91 specifying the length and capacity aren't strictly necessary in general. By specifying them, it allows the runtime to avoid extra allocations to grow and maybe copy the slice, thus creating less garbage. 

I'm not familiar with this part of the code, but it looks to me like the upper bound is an arbitrary batch size so that the tombstoned parts are deleted in chunks rather than all deleted at the end.

I have some not-especially-scientific benchmarks relating to slice reallocations in [this gist](https://gist.github.com/mark-rushakoff/b5650bd8f06bece0b9fd).